### PR TITLE
Add permissions handling to sambacc

### PR DESCRIPTION
**Pull request by phlogistonjohn**
These patches add general support for choosing and configuring a "permissions handler". Currently our handlers are very simple and allow setting permissions once, always, or never (none=no-op).

Then, in the code paths recently added to ensure share dirs exist we add lines to invoke the permissions handler. In most scenarios this means that the first time this code is called the permissions will be set. On subsequent calls the code will see the status xattr is set and skip changing the perms. As noted above, this is configurable.

### CODE REVIEW

File: sambacc/config.py
Comment (phlogistonjohn) [reply sambacc/config.py]: done.

File: sambacc/config.py
Comment (phlogistonjohn) [reply sambacc/config.py]: done

File: sambacc/config.py
Comment (phlogistonjohn) [reply sambacc/config.py:400]: Good point, but this is existing code, so I'll defer it for a future PR in which we can improve the type hinting generally.

File: sambacc/permissions.py
Comment (phlogistonjohn) [reply sambacc/permissions.py]: done.

File: tests/container/Containerfile
Comment (phlogistonjohn) [reply tests/container/Containerfile:14]: yep. I want to test some of this manually too, but it's easier to add the needed stuff to samba-containers after the code is merged in sambacc (otherwise you *have* to have a draft/wip pr in order to avoid accidentally using unmerged sambacc branches/commits).

I also am thinking we may want to add the ability to build sambacc as an RPM instead of a wheel. For the rpm based distros we use in samba-container this could avoid having to repeat ourselves wrt dependencies.

File: sambacc/config.py
Comment (spuiuk) [inline sambacc/config.py:384]: I wonder if leaving it to default to the no-op handler is safer. Let the operator explicitly mention that the permission needs to be changed. 

I think it is a good idea to leave the permissions alone if the user has provided a pvc unless the user has explicitly asked for permissions to be changed. When creating a new pvc from a pvc definition we can setup the permissions.

This can also be handled in a new PR.

**Review by spuiuk — APPROVED** [review]
Apart from that one comment.  ACK to the patchset.

I'll leave it to you to decide what the default handler should be.

File: sambacc/config.py
Comment (phlogistonjohn) [reply sambacc/config.py:384]: Yeah, I was a bit on the fence about what the default should be. It is very easy for the operator to simply enable the permissions initialization method, so I think may be best to leave it disabled for now.  Let's discuss that in a follow-up PR as you suggest.

**Comment by phlogistonjohn** [comment]
This has been updated to use "none" as the default permissions mode.

**Comment by phlogistonjohn** [comment]
I'd like to merge this if there are no objections. Last call for feedback on the latest versions - @spuiuk @synarete

**Review by spuiuk — APPROVED** [review]
Acked-by: sprabhu@redhat.com

**Comment by synarete** [comment]
> Acked-by: [sprabhu@redhat.com](mailto:sprabhu@redhat.com)

Acked-by: [ssharon@redhat.com](mailto:ssharon@redhat.com)

**Pull request merged**
